src/NUnitFramework/framework/Interfaces/FixtureMethod.cs

```diff
@@ -0,0 +1,105 @@
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using NUnit.Compatibility;
+
+namespace NUnit.Framework.Interfaces
+{
+    /// <summary>
+    /// Specifies a method declaration in the context of a certain point
+    /// in the fixture’s inheritance hierarchy.
+    /// </summary>
+    public struct FixtureMethod : IEquatable<FixtureMethod>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FixtureMethod"/> struct.
+        /// </summary>
+        public FixtureMethod(Type fixtureType, MethodInfo method)
+        {
+            Guard.ArgumentNotNull(fixtureType, nameof(fixtureType));
+            Guard.ArgumentNotNull(method, nameof(method));
+
+            if (!method.DeclaringType.GetTypeInfo().IsAssignableFrom(fixtureType))
+                throw new ArgumentException("The specified fixture type does not contain the specified method.");
+
+            FixtureType = fixtureType;
+            Method = method;
+        }
+
+        /// <summary>
+        /// May be more derived than <c>Method.DeclaringType</c>. Carries the
+        /// information that used to be held in the deprecated <c>MethodInfo.ReflectedType</c>.
+        /// </summary>
+        public Type FixtureType { get; }
+
+        /// <summary>
+        /// The method declaration.
+        /// </summary>
+        public MethodInfo Method { get; }
+
+        /// <summary>
+        /// Displays the specified fixture and method.
+        /// </summary>
+        public override string ToString()
+        {
+            var parameters = Method.GetParameters();
+            var parametersDisplay = new string[parameters.Length];
+
+            for (var i = 0; i < parametersDisplay.Length; i++)
+                parametersDisplay[i] = parameters[i].ToString();
+
+            return $"{FixtureType.FullName}.{Method.Name}({string.Join(", ", parametersDisplay)})";
+        }
+
+        /// <summary>
+        /// Indicates whether this instance and another instance are equal.
+        /// </summary>
+        public bool Equals(FixtureMethod other)
+        {
+            return FixtureType == other.FixtureType && Method == other.Method;
+        }
+
+        /// <summary>
+        /// Indicates whether this instance and a specified object are equal.
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            return obj is FixtureMethod && Equals((FixtureMethod)obj);
+        }
+
+        /// <summary>
+        /// Returns the hash code for this instance.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            var hashCode = 662238274;
```

Comment:
Can you enlighten me on this method of generating a hash code? What is the significance of the number? I see this in Roslyn, but initializing with -1923861349.

Reply:
This was me using Roslyn 15.6.5, Ctrl-dotting on the struct declaration and selecting "[Generate Equals and GetHashCode](https://docs.microsoft.com/visualstudio/ide/reference/generate-equals-gethashcode-methods)." I don't know why it picked the constants it did, though I'm curious why I can't find discussion on it. Here's the [evolution of the generating code](https://github.com/dotnet/roslyn/commits/master/src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions_CreateGetHashCodeMethod.cs) in Roslyn.

Reply:
@rprouse It looks like the initial hashcode is derived from the names of the members involved, so you'd expect it to be different most of the type: https://github.com/dotnet/roslyn/blob/b6f7194660ebf549038a0aecd9f3b4ad3dd4d0ac/src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions_CreateGetHashCodeMethod.cs#L63-L73

Reply:
Cyrus says he was just copying the logic used by the C# compiler to implement GetHashCode for anonymous types, so the discussion would have all happened in the development of C# 3. =D